lib/src/browser/disposable_browser.dart

```diff
@@ -12,11 +12,191 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
+import 'dart:async';
 import 'dart:html';
 
+import 'package:meta/meta.dart';
+import 'package:w_common/func.dart';
+
 import 'package:w_common/src/common/disposable.dart' as disposable_common;
 
-class Disposable extends disposable_common.Disposable {
+class _InnerDisposable extends disposable_common.Disposable {
```

Comment:
This is the class that will be instantiated internally in the browser version of Disposable and all calls will be proxied to it. It allows us to call `Disposable.onDispose` when `dispose()` is called.